test.js

```diff
@@ -198,3 +231,41 @@ test('grouped short-flags work', t => {
 	t.true(flags.c);
 	t.true(flags.l);
 });
+
+test('required flag', async t => {
+	const {stdout} = await execa('./fixture.js', [
+		'--meow',
+		'cat'
+	]);
+	t.is(stdout, 'Missing required option:\n  --foo, -o\n\nRun with --help to view help information.');
+});
+
+test('provisional require', async t => {
+	const {stdout: i} = await execa('./fixture.js', [
+		'--unicorn',
+		'cat',
+		'--foo',
+		'jump'
+	]);
+
+	const {stdout: f} = await execa('./fixture.js', [
+		'--meow',
+		'cat'
+	]);
+
+	t.is(i, 'Missing required option:\n  --fence, -f\n\nRun with --help to view help information.');
+	t.is(f, 'Missing required option:\n  --foo, -o\n\nRun with --help to view help information.');
```

Comment:
It would be better to test only the important part of the message, so other parts are free to change in the future without having to update all the tests. You can use `t.regex` instead.

All we need to test is `Missing required option:\n  --foo, -o`.